src/makeCalculator.js

```diff
@@ -4,7 +4,42 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+    divide(number) {
+      this.result /= number;
+
+      return this;
+    },
+    operate(callback, number) {
+      callback.call(this, number);
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
The function name 'makeCalculator' is appropriate as it describes the action of the function, which is in line with the naming convention suggested in the checklist. However, consider checking for any edge cases or error handling that might be needed, for example when dividing by zero.